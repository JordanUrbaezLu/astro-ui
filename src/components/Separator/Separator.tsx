import * as React from "react";
import classNames from "classnames";
import styles from "./Separator.module.scss";

export type SeparatorDirection = "horizontal" | "vertical";

export interface SeparatorProps
  extends React.ComponentPropsWithoutRef<"span"> {
  /**
   * The accessible label for the Separator
   */
  ariaLabel?: string;
  /**
   * The direction of the Separator
   *
   * @default "horizontal"
   */
  direction?: SeparatorDirection;
  /**
   * The stroke width of the Separator
   *
   * @default 1
   */
  stroke?: number;
}

/**
 * @public
 *
 * @description
 *
 * The Separator component is used to divide some content.
 */
export const Separator: React.FC<SeparatorProps> = (props) => {
  const {
    className,
    direction = "horizontal",
    stroke = 1,
    ...rest
  } = props;
  return (
    <span
      aria-label=""
      className={classNames(className, styles.separator)}
      role="separator"
      style={
        direction === "horizontal"
          ? { borderTop: `${stroke}px solid #c8c8c8`, width: "100%" }
          : {
              borderLeft: `${stroke}px solid #c8c8c8`,
              height: "100%",
            }
      }
      {...rest}
    />
  );
};

Separator.displayName = "Separator";
