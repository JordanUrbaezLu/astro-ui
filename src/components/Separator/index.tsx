/* istanbul ignore file */
export * from "./Separator";
