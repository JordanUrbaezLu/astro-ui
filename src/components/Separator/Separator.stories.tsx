import { FlexColumnContainer } from "../../../.storybook/components/FlexColumnContainer";
import { Separator } from "./Separator";

export default {
  component: Separator,
  title: "Components/Presentation/Separator",
};

export const Default = () => <Separator />;

export const Stroke = () => {
  return (
    <FlexColumnContainer>
      <Separator direction="horizontal" />
      <Separator direction="horizontal" stroke={2} />
      <Separator direction="horizontal" stroke={5} />
    </FlexColumnContainer>
  );
};

export const Horizontal = () => <Separator direction="horizontal" />;

export const Vertical = () => (
  <div style={{ height: "500px" }}>
    <Separator direction="vertical" />
  </div>
);
