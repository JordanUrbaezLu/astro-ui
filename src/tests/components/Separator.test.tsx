import { screen, render } from "@testing-library/react";
import { axe } from "jest-axe";
import { Separator } from "../../components/Separator/Separator";

describe("Separator", () => {
  test("Should render component correctly.", () => {
    const { container } = render(<Separator />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(<Separator />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test("Should render horizontally correctly.", () => {
    render(<Separator direction="horizontal" />);

    expect(screen.getByRole("separator")).toHaveStyle({
      borderTop: "1px solid #c8c8c8",
    });
  });
  test("Should render vertically correctly.", () => {
    render(<Separator direction="vertical" />);

    expect(screen.getByRole("separator")).toHaveStyle({
      borderLeft: "1px solid #c8c8c8",
    });
  });
});
