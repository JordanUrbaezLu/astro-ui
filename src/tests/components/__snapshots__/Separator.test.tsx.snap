// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator Should render component correctly. 1`] = `
<span
  aria-label=""
  class="separator"
  role="separator"
  style="border-top: 1px solid #c8c8c8; width: 100%;"
/>
`;
